mavros_extras/src/plugins/odom.cpp

```diff
@@ -63,96 +66,103 @@ class OdometryPlugin : public plugin::PluginBase {
 
 	MAV_ESTIMATOR_TYPE estimator_type;
 
+	//! child if for tf
+	std::string desired_frame;
+
 	/* -*- callbacks -*- */
 
 	void odom_cb(const nav_msgs::Odometry::ConstPtr &odom)
 	{
-		// lookup transforms
+		/** Lookup transforms
+		 *  @todo Implement in a more general fashion in the API IOT apply frame transforms
+		 *  This should also run in parallel on a thread
+		 */
 		Eigen::Affine3d tf_child2ned;
 		Eigen::Affine3d tf_parent2ned;
 		try {
+			// transform lookup WRT world frame
 			tf_child2ned = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
-				odom->child_frame_id, "local_origin_ned", ros::Time(0)));
+							odom->child_frame_id, "local_origin" + desired_frame, ros::Time(0)));
+			// transform lookup WRT body frame
 			tf_parent2ned = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
-				odom->header.frame_id, "local_origin_ned", ros::Time(0)));
+							odom->header.frame_id, "local_origin" + desired_frame, ros::Time(0)));
```

Comment:
this is currently resolved to "local_originned", so I think it would be better to just leave the full frame name in the config file and not assume the user wants local_origin something.

Reply:
oh sorry it's a typo.

This assumption will be a way of generalize it for all the API.